### PR TITLE
share pull secrets with db

### PR DESCRIPTION
**Pull request by gazarenkov**
## Description

Share spec.Application.ImagePullRequests with Local Database Pod

## Which issue(s) does this PR fix or relate to

https://issues.redhat.com/browse/RHIDP-1484

## PR acceptance criteria

- [x] Tests added

### CODE REVIEW

File: pkg/model/db-statefulset_test.go [hunk 1]
```diff
@@ -58,3 +55,40 @@ func TestOverrideDbImage(t *testing.T) {
...
+func TestImagePullSecretSpec(t *testing.T) {
+	bs := *dbStatefulSetBackstage.DeepCopy()
+	bs.Spec.Application.ImagePullSecrets = []string{"my-secret1", "my-secret2"}
+
+	testObj := createBackstageTest(bs).withDefaultConfig(true)
+	model, err := InitObjects(context.TODO(), bs, testObj.externalConfig, true, false, testObj.scheme)
+	assert.NoError(t, err)
+
+	assert.Equal(t, 2, len(model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets))
+	assert.Equal(t, "my-secret1", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[0].Name)
```
Comment (rm3l) [inline pkg/model/db-statefulset_test.go:69]:
```suggestion
	assert.Equal(t, "my-secret1", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[0].Name)
	assert.Equal(t, "my-secret2", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[1].Name)
```

File: pkg/model/db-statefulset_test.go [hunk 2]
```diff
@@ -58,3 +55,40 @@ func TestOverrideDbImage(t *testing.T) {
...
+	bs = *dbStatefulSetBackstage.DeepCopy()
+	testObj = createBackstageTest(bs).withDefaultConfig(true).
+		addToDefaultConfig("db-statefulset.yaml", "ips-deployment.yaml")
+
+	model, err = InitObjects(context.TODO(), bs, testObj.externalConfig, true, true, testObj.scheme)
+	assert.NoError(t, err)
+
+	// if imagepullsecrets not defined - default used
+	assert.Equal(t, 2, len(model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets))
+	assert.Equal(t, "ips1", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[0].Name)
```
Comment (rm3l) [inline pkg/model/db-statefulset_test.go:81]:
```suggestion
	assert.Equal(t, "ips1", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[0].Name)
	assert.Equal(t, "ips2", model.localDbStatefulSet.statefulSet.Spec.Template.Spec.ImagePullSecrets[1].Name)
```

File: pkg/model/deployment_test.go [hunk 3]
```diff
@@ -80,3 +80,29 @@ func TestOverrideBackstageImage(t *testing.T) {
...
+
+	testObj := createBackstageTest(bs).withDefaultConfig(true).
+		addToDefaultConfig("deployment.yaml", "ips-deployment.yaml")
+
+	model, err := InitObjects(context.TODO(), bs, testObj.externalConfig, true, true, testObj.scheme)
+	assert.NoError(t, err)
+
+	// if imagepullsecrets not defined - default used
+	assert.Equal(t, 2, len(model.backstageDeployment.deployment.Spec.Template.Spec.ImagePullSecrets))
+	assert.Equal(t, "ips1", model.backstageDeployment.deployment.Spec.Template.Spec.ImagePullSecrets[0].Name)
```
[the remark this answers is not in the document]
Comment (gazarenkov) [reply pkg/model/deployment_test.go:93]: outdated

File: pkg/model/db-statefulset_test.go (hunk 2 above)
Comment (gazarenkov) [reply pkg/model/db-statefulset_test.go:82]: added

File: pkg/model/db-statefulset_test.go (hunk 1 above)
Comment (gazarenkov) [reply pkg/model/db-statefulset_test.go:69]: added

**Pull request merged**